engine/src/main/java/org/hibernate/search/analyzer/impl/LuceneAnalyzerReference.java

```diff
@@ -16,20 +19,51 @@
  */
 public class LuceneAnalyzerReference implements AnalyzerReference {
 
-	private final Analyzer analyzer;
+	private static final Log LOG = LoggerFactory.make();
+
+	private String name;
+
+	private Analyzer analyzer;
+
+	public LuceneAnalyzerReference(String name) {
+		this.name = name;
+		this.analyzer = null; // Not initialized
+	}
 
 	public LuceneAnalyzerReference(Analyzer analyzer) {
+		this.name = null;
 		this.analyzer = analyzer;
 	}
 
 	@Override
+	public String getAnalyzerName() {
+		return name;
+	}
+
+	@Override
 	public Analyzer getAnalyzer() {
+		if ( analyzer == null ) {
+			throw LOG.lazyLuceneAnalyzerReferenceNotInitialized( this );
+		}
 		return analyzer;
 	}
 
+	public boolean isInitialized() {
+		return analyzer != null;
+	}
+
+	public void initialize(Analyzer analyzer) {
+		if ( this.analyzer != null ) {
+			throw new AssertionFailure( "An analyzer reference has been initialized more than once." );
```

Comment:
I think we should include the name for the builtin analyzers. That would be more clear. (I know this code has moved so it's probably something to apply on top of the other changes)